.github/workflows/release.yml

```diff
@@ -1,461 +1,192 @@
 
-  publish-binaries:
-    name: Publish Binaries
-    needs: [lint, build, test-linux, test-macos, test-windows, versions]
-    uses: ./.github/workflows/publish-binaries.yml
-    with:
-      goreleaser-config: .goreleaser.yml
-      goreleaser-flags: -p 3 --release-notes=CHANGELOG_PENDING.md
-    secrets:
-      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
-      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
-      AWS_UPLOAD_ROLE_ARN: ${{ secrets.AWS_UPLOAD_ROLE_ARN }}
-      INPUT_GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
-      GORELEASER_KEY: ${{ secrets.GORELEASER_KEY }}
-
-  lint:
-    # See https://github.com/pulumi/pulumi/issues/9280 for why this is set to v1.44
-    container: golangci/golangci-lint:v1.44
-    name: Lint ${{ matrix.directory }}
-    strategy:
-      matrix:
-        directory: [sdk, pkg, tests]
+  s3-blobs:
+    name: s3 blobs
     runs-on: ubuntu-latest
     steps:
       - name: Checkout Repo
-        uses: actions/checkout@v2
-      - name: Lint ${{ matrix.directory }}
-        run: |
-          cd ${{ matrix.directory }} && golangci-lint run -c ../.golangci.yml
-  language-sdk-lint:
-    name: Lint Language SDKs
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-region: us-east-2
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-duration-seconds: 3600
+          role-external-id: upload-pulumi-release
+          role-session-name: pulumi@githubActions
+          role-to-assume: ${{ secrets.AWS_UPLOAD_ROLE_ARN }}
+      - name: Download release artifacts
+        run: |
+          mkdir -p artifacts
+          gh release download "v${PULUMI_VERSION}" --dir ./artifacts --pattern 'pulumi-*'
+          find artifacts
+      - name: Publish Blobs
+        run: |
+          aws s3 sync artifacts s3://get.pulumi.com/releases/sdk --acl public-read
+
+  pr:
+    # Relies on the Go SDK being published to update pkg
+    name: PR
+    needs: [sdks]
+    uses: ./.github/workflows/release-pr.yml
+    permissions:
+      contents: write
+      pull-requests: write
+    with:
+      ref: ${{ inputs.ref }}
+      version: ${{ inputs.version }}
+      next-version: ${{ inputs.next-version }}
+      release-notes: ${{ inputs.release-notes }}
+      queue-merge: false
+    secrets: inherit
+
+  dispatch:
+    name: ${{ matrix.job.name }}
+    if: inputs.run-dispatch-commands && !contains(inputs.version, '-')
+    runs-on: ubuntu-latest
+    needs: [pr]
     strategy:
+      fail-fast: false
       matrix:
-        go-version: [1.17.x]
-        python-version: [3.9.x]
-        dotnet-version: [6.0.x]
-        node-version: [14.x]
-    runs-on: ubuntu-latest
+        job:
+          - name: Templates Smoke Test
```

Comment:
Has this action run? How do we test run this logic, or perhaps a subset of it?  

I'm assuming using a pre-release release through official channels is not an option.

We could try to test-drive against `<forked>/examples` etc, maybe comment out winget-deploy and choco-deploy, whatever is not easy to run in forked mode.

Reply:
I'm not sure - I think triggering examples, templates, etc tests on PRs and during merge queues is out of scope.

It has definitely run: https://github.com/pulumi/pulumi/runs/8166609012?check_suite_focus=true

Post-release, would you like to see an issue created to:
* Fixing examples & templates tests to pass reliably.
* For those & others, enabling testing of our post-release actions *during* the merge queue workflow? That is, with a snapshot build, or on a cron?

Reply:
I think we converged on trying this out in-place using a real point release of pulumi/pulumi  - my biggest concern was just that we have some logic in the yamls here that might have never been executed anywhere.

These issues sound helpful yes, in the area/build backlog. We can later pick up a subset of examples/tests to verify against.